### ISSUE TITLE
[DEMO]

### ISSUE THREAD

**Issue by nsansen** [PennyLaneAI/qml#372]
#### General information

**Name**
Nikolaos Schetakis (nikschet)

**Affiliation** (optional)
1. Alma-Sistemi srl
2. Technical University of Crete

**Twitter** (optional)
Your Twitter username, if interested; helps us advertise your demo while linking directly back to you.

**Image** (optional)
Suggested image to use when advertising your demo on Twitter; can be provided via hyperlink or by copy/pasting directly in GitHub.

--------------------------------------------------------------------------------

#### Demo information

**Title**
A proposed hybrid NN using Data-Reuploading technique 

**Abstract**
By combining a standar Variational Classifier (VC) with a Data-Reuploading Classifier (DRC) we create a new Qnode which is then used as a quantum layer in a Hybrid NN. 

**Relevant links**
Actual code: https://github.com/nsansen/Quantum-Machine-Learning/blob/main/Pennylane%20DEMO%20v2.ipynb
Please check and cite this paper for more information: https://arxiv.org/submit/4012476/view

Variational classifier demo: https://pennylane.ai/qml/demos/tutorial_variational_classifier.html
Data reuploading classifier demo:  https://pennylane.ai/qml/demos/tutorial_data_reuploading_classifier.html
Turning quantum nodes into Keras layer: https://pennylane.ai/qml/demos/tutorial_qnn_module_tf.html

**Comment by CatalinaAlbornoz** [comment]
Thank you for submitting this demo @nsansen! I will review and hopefully have it up in the website very soon.
Are you participating in the [PennyLane Quantum Carnival](https://pennylane.ai/carnival)? This demo will give you tickets.

**Comment by nsansen** [comment]
Hello, I just subscribed to the quantum carnival. And plan to explore it.

Best
Nikolaos Schetakis


Best regards,
Nikolaos Schetakis

LEGAL DISCLAIMER:
The contents of this email and any transmitted files are confidential and intended solely for the use of the individual or entity to whom they are addressed. We hereby exclude any warranty and any liability as to the quality or accuracy of the contents of this email and any attached transmitted files. If you are not the intended recipient, be advised that you have received this email in error and that any use, dissemination, forwarding, printing or copying of this email is strictly prohibited. If you have received this email in error please contact the sender and delete the material from any computer.

**Comment by CatalinaAlbornoz** [comment]
Awsome @nsansen! Did you register with the same email as the one you have on the demo? If not you can let me know on a private message on Slack.

**Comment by nsansen** [comment]
Hello , is there a pennylane slack workspace? Please send me the name. I think I register under the: ***@***.*** ***@***.***> email.

Best regards,
Nikolaos Schetakis

LEGAL DISCLAIMER:
The contents of this email and any transmitted files are confidential and intended solely for the use of the individual or entity to whom they are addressed. We hereby exclude any warranty and any liability as to the quality or accuracy of the contents of this email and any attached transmitted files. If you are not the intended recipient, be advised that you have received this email in error and that any use, dissemination, forwarding, printing or copying of this email is strictly prohibited. If you have received this email in error please contact the sender and delete the material from any computer.

**Comment by CatalinaAlbornoz** [comment]
Hi @nsansen, yes! This is the link to the xanadu slack https://xanadu-quantum.slack.com  . There you will find a PennyLane channel and others as well. Please send me the email you registered on a private message on Slack. Thanks!

**Comment by nsansen** [comment]
Dear ,

On Monday the paper will be published on arXiv, I will provide a link to you then, since I am not sure if I can edit the demo.

Thanks!!!


Best regards,
Nikolaos Schetakis

LEGAL DISCLAIMER:
The contents of this email and any transmitted files are confidential and intended solely for the use of the individual or entity to whom they are addressed. We hereby exclude any warranty and any liability as to the quality or accuracy of the contents of this email and any attached transmitted files. If you are not the intended recipient, be advised that you have received this email in error and that any use, dissemination, forwarding, printing or copying of this email is strictly prohibited. If you have received this email in error please contact the sender and delete the material from any computer.

**Comment by nsansen** [comment]
Hello!

The paper got published in the arXiv. Please if you upload the demo use this link to cite the paper (and not the old one I have written) : https://arxiv.org/pdf/2111.03372.pdf

Many thanks!!!


Best regards,
Nikolaos Schetakis
Best regards,
Nikolaos Schetakis

LEGAL DISCLAIMER:
The contents of this email and any transmitted files are confidential and intended solely for the use of the individual or entity to whom they are addressed. We hereby exclude any warranty and any liability as to the quality or accuracy of the contents of this email and any attached transmitted files. If you are not the intended recipient, be advised that you have received this email in error and that any use, dissemination, forwarding, printing or copying of this email is strictly prohibited. If you have received this email in error please contact the sender and delete the material from any computer.

**Comment by CatalinaAlbornoz** [comment]
Hi @nsansen, I managed to make it work with Python 3.8.10. 
I still see several warnings so I'll look into it tomorrow to see if they can be fixed.

**Comment by nsansen** [comment]
Thank you very much!! Let me know if you need me to do anything.


Best regards,
Nikolaos Schetakis

LEGAL DISCLAIMER:
The contents of this email and any transmitted files are confidential and intended solely for the use of the individual or entity to whom they are addressed. We hereby exclude any warranty and any liability as to the quality or accuracy of the contents of this email and any attached transmitted files. If you are not the intended recipient, be advised that you have received this email in error and that any use, dissemination, forwarding, printing or copying of this email is strictly prohibited. If you have received this email in error please contact the sender and delete the material from any computer.

**Comment by CatalinaAlbornoz** [comment]
Hi @nsansen ,
The changes that should be made are:

In section "3. Build Quantum node"
 - Comment `qml.enable_tape()`
 - Change `grad_method="backprop"` to `diff_method="backprop"`

In section 5. Make predictions and plot training history + prediction grids
- Change `modelh.predict_proba(X_test)` to `modelh.predict(X_test)`
- Change `modelh.predict_classes(X_test)` to `(modelh.predict(X_test) > 0.5).astype("int32")`
- Change `modelh.predict_classes(grid)` to `(modelh.predict(grid) > 0.5).astype("int32")`

You can make these changes in the document that you already have. This will remove the warnings I was seeing.
If this works on your side just let me know and I can push the demo card into the community demos page!

**Comment by nsansen** [comment]
Dear Catalina,  Thank you very much for your time and effort. I made this changes and I get no warning. Everything works great! I found out a small mistake I made regarding numbering the sections so section "3. Build Quantum node" should be "5. Build Quantum node" and the other numbers need some rearrangement. I fix these and the code is the one attached. BestNikolaos Schetakis Sent from Mail for Windows From: CatalinaAlbornozSent: 10 November 2021 23:20To: PennyLaneAI/qmlCc: Nikolaos Schetakis; MentionSubject: Re: [PennyLaneAI/qml] [DEMO] (Issue #372) Hi @nsansen ,The changes that should be made are:In section "3. Build Quantum node"Comment qml.enable_tape()Change grad_method="backprop" to diff_method="backprop"In section 5. Make predictions and plot training history + prediction gridsChange modelh.predict_proba(X_test) to modelh.predict(X_test)Change modelh.predict_classes(X_test) to (modelh.predict(X_test) > 0.5).astype("int32")Change modelh.predict_classes(grid) to (modelh.predict(grid) > 0.5).astype("int32")You can make these changes in the document that you already have. This will remove the warnings I was seeing.If this works on your side just let me know and I can push the demo card into the community demos page!—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by CatalinaAlbornoz** [comment]
Hi @nsansen, good catch with the numbering. Remember to upload the updated version of the notebook [here](https://github.com/nsansen/Quantum-Machine-Learning/blob/main/Pennylane%20DEMO%20v2.ipynb).

**Comment by nsansen** [comment]
Hi! Updated version is uploaded in Github  (Pennylane DEMO v4.ipynb) https://github.com/nsansen/Quantum-Machine-Learning/blob/main/Pennylane%20DEMO%20v4.ipynb Many thanks. If you need me to do anything else let me know BestNikolaos Schetakis  Sent from Mail for Windows From: CatalinaAlbornozSent: 11 November 2021 16:28To: PennyLaneAI/qmlCc: Nikolaos Schetakis; MentionSubject: Re: [PennyLaneAI/qml] [DEMO] (Issue #372) Hi @nsansen, good catch with the numbering. Remember to upload the updated version of the notebook here.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by CatalinaAlbornoz** [comment]
Thank you @nsansen! I think we're all good! I believe this should be up in the page tomorrow. Congrats! Do you want us to tag you in the tweet where we will promote it?

**Comment by nsansen** [comment]
Thank you very much!!

Sorry I don’t have an official twitter account.



Best regards,
Nikolaos Schetakis

LEGAL DISCLAIMER:
The contents of this email and any transmitted files are confidential and intended solely for the use of the individual or entity to whom they are addressed. We hereby exclude any warranty and any liability as to the quality or accuracy of the contents of this email and any attached transmitted files. If you are not the intended recipient, be advised that you have received this email in error and that any use, dissemination, forwarding, printing or copying of this email is strictly prohibited. If you have received this email in error please contact the sender and delete the material from any computer.

**Comment by CatalinaAlbornoz** [comment]
No problem @nsansen! Just one last comment, if you could add a note on the top of the notebook saying that this only runs with Python 3.7-3.8 it would be great (without changing the notebook URL). The Demo should be up in the page in the next few hours!

**Comment by nsansen** [comment]
Ok the change has been implemented as requested. BestNS Sent from Mail for Windows From: CatalinaAlbornozSent: 12 November 2021 16:34To: PennyLaneAI/qmlCc: Nikolaos Schetakis; MentionSubject: Re: [PennyLaneAI/qml] [DEMO] (Issue #372) No problem @nsansen! Just one last comment, if you could add a note on the top of the notebook saying that this only runs with Python 3.7-3.8 it would be great (without changing the notebook URL). The Demo should be up in the page in the next few hours!—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by CatalinaAlbornoz** [comment]
The demo is up! Congratulations @nsansen! You can find it here https://pennylane.ai/qml/demos_community.html. Feel free to share it on social media!

**Comment by nsansen** [comment]
Dear,

Is it possible to update the paper Link in the DEMO : Hybrid Neural Network using Data-Reuploading technique ? (By Nikolaos Schetakis 11/04/2021) 

The paper was accepted by Scientific reports and the link is here: https://link.springer.com/content/pdf/10.1038/s41598-022-14876-6.pdf

Thank you in advance!

Best regards,
Nikolaos Schetakis

LEGAL DISCLAIMER:
The contents of this email and any transmitted files are confidential and intended solely for the use of the individual or entity to whom they are addressed. We hereby exclude any warranty and any liability as to the quality or accuracy of the contents of this email and any attached transmitted files. If you are not the intended recipient, be advised that you have received this email in error and that any use, dissemination, forwarding, printing or copying of this email is strictly prohibited. If you have received this email in error please contact the sender and delete the material from any computer.

**Comment by ikurecic** [comment]
Dear @nsansen,

Of course, that's not a problem — thank you for letting us know!
You should be able to see the change soon.

All the best,
Ivana
